src/models/RewardsVault.ts

```diff
@@ -0,0 +1,34 @@
+import { Address } from '@graphprotocol/graph-ts'
+import { RewardsVault as RewardsVaultContract } from '../../generated/templates/RewardsVault/RewardsVault'
+import { RewardsVault } from '../../generated/schema'
+
+export function getOrCreateRewardsVault(
+  contractAddress: Address,
+): RewardsVault {
+  let id = contractAddress.toHexString()
+
+  let rewardsVault = RewardsVault.load(id)
+
+  if (rewardsVault != null) {
+    return rewardsVault as RewardsVault
+  }
+
+  {
+    let contract = RewardsVaultContract.bind(contractAddress)
+
+    let rewardsVault = new RewardsVault(id)
+
+    rewardsVault.allRewardsUnlocked = false
+    rewardsVault.lockupPeriods = contract.LOCKUP_PERIODS().toI32()
+    rewardsVault.period = contract.PERIOD().toI32()
+    rewardsVault.vaultStartTime = contract.vaultStartTime().toI32()
+    rewardsVault.stakingRewardsContract = contractAddress.toHexString()
```

Comment:
The 'rewardsVault' is a singleton, so shouldn't be directly back linked to an individual staking rewards contract - anyone can deposit here technically